[Feature] Add RandomCenterCrop into transform.py

zhangjinCV opened the pull request:
Add a RandomCenterCrop into transform.py. I had made some simple tests for the error input.

zhangjinCV committed:
add RandomCenterCrop into transform.py

Add a RandomCenterCrop into transform, I have made some simple tests for the error input.

wuyefeilin commented on paddleseg/transforms/transforms.py:
```diff
@@ -699,6 +699,86 @@ def __call__(self, im, label=None):
+    def __call__(self, im, label=None):
+        """
+        Args:
+            crop_ration (tuple or list or float, optional): Clipping range. Default: (0.8-1).
```
参数应该写im, label，具体可以参考一下其他的算子

wuyefeilin commented on paddleseg/transforms/transforms.py:
```diff
@@ -699,6 +699,86 @@ def __call__(self, im, label=None):
+    Crops the given the input data at the center.
+
+    Args:
+        crop_ration (tuple or list or float, optional): Clipping range. Default: 0.8.
```
crop_ration 应该h和w分别设置，分别提供h, w的随机比例范围。 你这边是[0, crop_ration]容易产生过小的结果，甚至是0*0。

建议参数可以设置成h_ration, w_ration, 取值为(low, high), 比如可以取（0.5, 0.8)，然后在这个范围内随机进行裁剪

另外如果可以的话， 可以增加一下大小范围的随机裁剪如（500， 800）。这样生成的图片更方便后续的处理。参数可以取h_value, w_value哈

michaelowenliu commented:
@zhangjin12138 Could you please refine your pr referring to the comments? btw, please submit pr into the develop branch first, and then cherry pick to release/2.3, thanks.

zhangjinCV commented:
Sorry, I have been busy with paper submission and academic matters during this period. I will complete this pr in two days.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "PaddlePaddle/PaddleSeg"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年11月10日(星期三) 中午11:23
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [PaddlePaddle/PaddleSeg] add RandomCenterCrop into transform.py (PR #1475)





 
@zhangjin12138 Could you please refine your pr referring to the comments? btw, please submit pr into the develop branch first, and then cherry pick to release/2.3, thanks.
 
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

zhangjinCV committed:
Merge branch 'PaddlePaddle:release/2.3' into pr_branch

zhangjinCV committed:
Update transforms.py

1. 参数应该写im, label，具体可以参考一下其他的算子
我检查了一下代码，没有找到其他参数不是im和label的。可能我有遗漏，麻烦帮我指出一下

2. crop_ration 应该h和w分别设置，分别提供h, w的随机比例范围。 你这边是[0, crop_ration]容易产生过小的结果，甚至是0*0。
我的本意是可接收的输入是float tuple or list  如果是float则h和w都裁剪至crop_ration到1的范围。按照这个要求，输入只接受tuple or list, 裁剪完的图片是尺寸是h * [crop_ration ,1] 这个里面的随机一个数,  w * [crop_ration ,1] 这个里面的随机一个数。不会出现0*0的情况。

3. 建议参数可以设置成h_ration, w_ration, 取值为(low, high), 比如可以取（0.5, 0.8)，然后在这个范围内随机进行裁剪
我将参数设置为list了，接收(h_ration, w_ration)。裁剪范围是[0.5-1]。因为我觉得不是必须要裁剪，尽量也要有原图的大小。

4. 另外如果可以的话， 可以增加一下大小范围的随机裁剪如（500， 800）。这样生成的图片更方便后续的处理。参数可以取h_value, w_value哈
我没有太理解这个意思，按我所接触的随即裁剪，只能按比例剪去边。如果觉得剪掉的图像太小，则需要Resize 操作。一般来说，随即中心裁剪后面需要有一个Resize操作，而不是Resize之后再crop。因为Resize的时候就已经丢失了图片的部分精度。

zhangjinCV commented:
我已经修改了部分代码，麻烦您看一下。这是我针对提出问题的反馈

1. 参数应该写im, label，具体可以参考一下其他的算子
我检查了一下代码，没有找到其他参数不是im和label的。可能我有遗漏，麻烦帮我指出一下

2. crop_ration 应该h和w分别设置，分别提供h, w的随机比例范围。 你这边是[0, crop_ration]容易产生过小的结果，甚至是0*0。
我的本意是可接收的输入是float tuple or list  如果是float则h和w都裁剪至crop_ration到1的范围。按照这个要求，输入只接受tuple or list, 裁剪完的图片是尺寸是h * [crop_ration ,1] 这个里面的随机一个数,  w * [crop_ration ,1] 这个里面的随机一个数。不会出现0*0的情况。

3. 建议参数可以设置成h_ration, w_ration, 取值为(low, high), 比如可以取（0.5, 0.8)，然后在这个范围内随机进行裁剪
我将参数设置为list了，接收(h_ration, w_ration)。裁剪范围是[0.5-1]。因为我觉得不是必须要裁剪，尽量也要有原图的大小。

4. 另外如果可以的话， 可以增加一下大小范围的随机裁剪如（500， 800）。这样生成的图片更方便后续的处理。参数可以取h_value, w_value哈
我没有太理解这个意思，按我所接触的随即裁剪，只能按比例剪去边。如果觉得剪掉的图像太小，则需要Resize 操作。一般来说，随即中心裁剪后面需要有一个Resize操作，而不是Resize之后再crop。因为Resize的时候就已经丢失了图片的部分精度。

wuyefeilin commented on paddleseg/transforms/transforms.py:
```diff
@@ -699,6 +699,71 @@ def __call__(self, im, label=None):
+    def __call__(self, im, label=None):
+        """
+        Args:
+            crop_ration (tuple or list or float, optional): Clipping range. Default: (0.5-1).
```
__call__的参数注释是对im, label这两个参数进行注释，而不是crop_ration

wuyefeilin commented on paddleseg/transforms/transforms.py:
```diff
@@ -699,6 +699,71 @@ def __call__(self, im, label=None):
+    """
+    Crops the given the input data at the center.
+    Args:
+        crop_ration (tuple or list, optional): Clipping range. Default:the crop range is 0.5-1.
```
这个参数的作用描述全一点哈，说明一下必须长度为2的list或tuple, 第一个是w的，第二个是h的。 看代码，给的值是最小的裁剪比例，这个需要注明一下

wuyefeilin reviewed (changes requested):
The annotation need improvement

zhangjinCV committed:
Merge branch 'PaddlePaddle:release/2.3' into pr_branch

zhangjinCV committed:
Update transforms.py

非常抱歉，因为我的变量起名引发了一些误会，原先的crop_ratio字面意思为裁剪比例。会让人以为是裁剪掉多少，实际我的意思应该是保留比例，所以出了第一次pr反馈的那个裁剪比例大的反馈。我更改了这个变量的名字，变为retain_ratio，意思为保留比例（功能不变）。另外按照这次的反馈添加了详细的注释解释。如果您觉得retain_ratio这个单词不好，我可以想想换成别的单词。再次因为我的失误导致你们看代码有困惑表示抱歉。

zhangjinCV commented:
非常抱歉，因为我的变量起名引发了一些误会，原先的crop_ratio字面意思为裁剪比例。会让人以为是裁剪掉多少，实际我的意思应该是保留比例，所以出了第一次pr反馈的那个裁剪比例大的反馈。我更改了这个变量的名字，变为retain_ratio，意思为保留比例（功能不变）。另外根据第二次反馈添加了详细的注释解释。如果您觉得retain_ratio这个单词不好，我可以想想换成别的单词。再次因为我的失误导致你们看代码有困惑表示抱歉。QQ:934399699 。这是我的联系方式. 如果有其它的反馈还可以通过qq联系我

wuyefeilin reviewed (approved):
LGTM

wuyefeilin commented:
> 非常抱歉，因为我的变量起名引发了一些误会，原先的crop_ratio字面意思为裁剪比例。会让人以为是裁剪掉多少，实际我的意思应该是保留比例，所以出了第一次pr反馈的那个裁剪比例大的反馈。我更改了这个变量的名字，变为retain_ratio，意思为保留比例（功能不变）。另外根据第二次反馈添加了详细的注释解释。如果您觉得retain_ratio这个单词不好，我可以想想换成别的单词。再次因为我的失误导致你们看代码有困惑表示抱歉。QQ:934399699 。这是我的联系方式. 如果有其它的反馈还可以通过qq联系我

感谢您的贡献，非常欢迎您继续为PaddleSeg添砖加瓦。

zhangjinCV commented:
谢谢！




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "PaddlePaddle/PaddleSeg"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年11月14日(星期天) 晚上6:19
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [PaddlePaddle/PaddleSeg] [Feature] add RandomCenterCrop into transform.py (PR #1475)





  
非常抱歉，因为我的变量起名引发了一些误会，原先的crop_ratio字面意思为裁剪比例。会让人以为是裁剪掉多少，实际我的意思应该是保留比例，所以出了第一次pr反馈的那个裁剪比例大的反馈。我更改了这个变量的名字，变为retain_ratio，意思为保留比例（功能不变）。另外根据第二次反馈添加了详细的注释解释。如果您觉得retain_ratio这个单词不好，我可以想想换成别的单词。再次因为我的失误导致你们看代码有困惑表示抱歉。QQ:934399699 。这是我的联系方式. 如果有其它的反馈还可以通过qq联系我
  
感谢您的贡献，非常欢迎您继续为PaddleSeg添砖加瓦。
 
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

michaelowenliu commented:
@zhangjin12138 嗨，非常感谢您的贡献，我们会将您的pr纳入PaddleSeg2.4的重要feature发版，能麻烦你将这个pr提交到develop分支吗？
